contracts/chain-adapters/Rerouter_Adapter.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { AdapterInterface } from "./interfaces/AdapterInterface.sol";
+
+/**
+ * @notice Contract containing logic to send messages from L1 to a target (not necessarily a spoke pool) on L2.
```

Comment:
Would be useful to note that one of these adapters should be deployed per L3. Also this adapter be mapped via HubPool to an L3 chain and actually, the L3 spoke pool kind of gets thrown away.

I think we should separately a commit an architecture diagram somewhere in the repo. Maybe something like [mermaid](https://github.blog/developer-skills/github/include-diagrams-markdown-files-mermaid/)

Reply:
Comment updated: https://github.com/across-protocol/contracts/pull/607/commits/cc386a73aeb64319c413c171e677b2465502f431

Also, diagramming something sounds good to me :+1: